src/app/static/src/app/store/modelRun/mutations.ts

```diff
@@ -1,46 +1,55 @@
+import {MutationTree} from "vuex";
 import {ModelRunState} from "./modelRun";
 import {PayloadWithType} from "../../types";
 import {ModelResultResponse, ModelStatusResponse, ModelSubmitResponse} from "../../generated";
-import {readyStateMutations} from "../shared/readyStateMutations";
 
-type ModelRunMutation = Mutation<ModelRunState>
-
-export interface ModelRunMutations {
-    ModelRunStarted: ModelRunMutation
-    RunStatusUpdated: ModelRunMutation,
-    PollingForStatusStarted: ModelRunMutation,
-    RunResultFetched: ModelRunMutation,
-    RunResultError: ModelRunMutation,
-    Ready: ModelRunMutation
+export enum ModelRunMutation {
+    ModelRunStarted="ModelRunStarted",
+    ModelRunError="ModelRunError",
+    RunStatusUpdated = "RunStatusUpdated",
+    PollingForStatusStarted = "PollingForStatusStarted",
+    RunResultFetched = "RunResultFetched",
+    RunResultError= "RunResultError",
+    RunStatusError="RunStatusError",
+    Ready= "Ready"
 }
 
-export const mutations: MutationTree<ModelRunState> & ModelRunMutations = {
-    ModelRunStarted(state: ModelRunState, action: PayloadWithType<ModelSubmitResponse>) {
+export const mutations: MutationTree<ModelRunState>  = {
+    [ModelRunMutation.ModelRunStarted](state: ModelRunState, action: PayloadWithType<ModelSubmitResponse>) {
         state.modelRunId = action.payload.id;
         state.status = {id: action.payload.id} as ModelStatusResponse;
         state.errors = [];
     },
 
-    RunStatusUpdated(state: ModelRunState, action: PayloadWithType<ModelStatusResponse>) {
+    [ModelRunMutation.RunStatusUpdated](state: ModelRunState, action: PayloadWithType<ModelStatusResponse>) {
         if (action.payload.done) {
             clearInterval(state.statusPollId);
             state.statusPollId = -1;
         }
         state.status = action.payload;
     },
 
-    PollingForStatusStarted(state: ModelRunState, action: PayloadWithType<number>) {
+    [ModelRunMutation.PollingForStatusStarted](state: ModelRunState, action: PayloadWithType<number>) {
         state.statusPollId = action.payload;
     },
 
-    RunResultFetched(state: ModelRunState, action: PayloadWithType<ModelResultResponse>) {
+    [ModelRunMutation.RunResultFetched](state: ModelRunState, action: PayloadWithType<ModelResultResponse>) {
         state.result = action.payload;
     },
 
-    RunResultError(state: ModelRunState, action: PayloadWithType<string>) {
+    [ModelRunMutation.RunResultError](state: ModelRunState, action: PayloadWithType<string>) {
         state.errors.push(action.payload);
     },
 
-    ...readyStateMutations
+    [ModelRunMutation.Ready](state: ModelRunState) {
+        state.ready = true;
+    },
+
+    [ModelRunMutation.ModelRunError](state: ModelRunState, action: PayloadWithType<string>) {
+        state.errors.push(action.payload);
+    },
+
+    [ModelRunMutation.RunStatusError](state: ModelRunState, action: PayloadWithType<string>) {
+        state.errors.push(action.payload);
```

Comment:
Maybe we need to be careful about this - if something goes wrong and we start getting an error on every poll request, this array is going to get huge with no obvious way to stop it. Should we stop polling if we get an error? I don't think we do at the moment.

Reply:
As thing stand, if you got one error during polling then the run successfully completed sometime later, the user wouldn't be able to proceed to subsequent steps anyway, as there would be an error in the list.

Reply:
Good point, not sure how to handle this. If the status endpoint errors, it might actually go on indefinitely. But we also don't want to halt the polling on a single failure. Maybe we need to do something clever like if we get 5 consecutive errors, halt and report the last one? I might just revert this here and leave for the subsequent ticket about reporting model run errors if that's ok?

Reply:
That's fine, I'm doing that ticket now and there's not much else to it. Will put in 'five strikes and you're out' into that. And I guess if you get polling errors but then it recovers, it will clear the errors so that the stepper can proceed.